### ISSUE TITLE
Filtering/Error Correction/Merging for poor quality paired-end read data

### ISSUE THREAD

**Issue by JacobRPrice** [benjjneb/dada2#176]
I'm attempting to apply the dada pipeline (as described in doi:10.12688/f1000research.8986.2) to a dataset we recently received and which, unfortunately, has a significant dropoff in quality early on in both Read 1/forward and Read 2/reverse (a representative example of the quality profile plot is included below). The samples were sequenced 2x300 bp on MiSeq. 

![screen shot 2017-01-30 at 12 28 28 pm](https://cloud.githubusercontent.com/assets/11934394/22442757/68a973d2-e70a-11e6-8b83-92eae6d339bc.png)
![screen shot 2017-01-30 at 12 48 07 pm](https://cloud.githubusercontent.com/assets/11934394/22442758/68bfb8e0-e70a-11e6-8ab7-7d2547c2e841.png)

Our primary challenge with this dataset is that we are torn between balancing quality of the overlap with the amount of overlap, especially since many of the reverse reads are of such low quality (5<Q<15 after base 100 or so) that one may be able to argue to simply use the forward reads alone.

We used the EMP 515F/926R primers to target V4 and V5 on 16S; we expected to have approximately 230 bases overlapping (on the untrimmed reads). We've previously trimmed reads when they drop below Q30-Q27 but as you can see, doing so would result in very short non-overlapping reads.

A colleague @sw1 and I have come up with a couple of alternatives and we wanted to hear if you  had any insight on how to handle this particular dataset. 

1. Trim both R1 and R2 as usual (resulting in no overlaps), proceed with the pipeline, and merge the reads with justContacenate=TRUE (as outlined in issue #72).

2. Use just R1 for each sample which may be sufficient, especially if we keep in mind the limited power of any taxanomic assignments (described in doi:10.1111/nph.12923). 

3. Approach trimming/filtering less conservatively, which is much more permissive than what I've done in the past for dada2. This may (will) significantly increase the run time during dada(). The upside of this is that we can play around with the number of mismatches allowed in mergePairs. We're thinking that the amount overlap may be more important than the quality of overlap, since mergePairs may act as a second screen. Also, while the Q scores are lower than we'd like, they still above the default phred score used in QIIME split libraries command.

It comes down to how we prioritize (a) overlapping (and longer) reads and (b) the amount of error we are inputting into the error estimation step. 

Thank you for any insight you'd be able to provide.

Jake

**Comment by joey711** [comment]
@JacobRPrice there are a lot of technical details one could jump into here, but there's a high-level question you may want to consider first. You expected 230 bp length, but you have about 90-100 bp reasonable quality from R1, maybe worse with R2. Something bad happened in the sequencer, or with your amplicons prior to going in the sequencer. Are you sure the amplification worked the way you expected? Those primers in this protocol usual result in a much different, persistently high quality profile. If you take the good-quality portion of some of these sequences and BLAST them, do you get 16S back? 

As for your questions and glancing at this quality profile, I'd guess that option (3) is probably not going to work very well. The chance of getting perfect overlap in this scenario is going to be very low, which means that you'll probably throw out a lot of read-pairs, even if you relax the quality thresholds to the point that reads at that length get through. 

I'd also point out that R1 is only slightly better than R2, which is to say that they're both pretty suspect. If this came from a sequencing provider, I'd ask them to re-do the sequencing at their cost, after they've double-checked their own QC measures at each step to see if they already know what went wrong. A QC on the amplicons, especially amplicon length profile and DNA quality also seems very reasonable here. 

If you don't have the option of resequencing and you think the high-quality regions are reliable, then options 1 or 2 seem worth considering. As you can tell by the rest of my response, though, I recommend quite a bit of caution. In the long run, you might save more time and resources simply re-doing the amplification and/or sequencing than trying to do the delicate bioinformatics gymnastics and interpretation that you've proposed.

@benjjneb , have any interesting wrinkle to add here?

**Comment by JacobRPrice** [comment]
@joey711, thank-you for your input. 

The template for these amplicons were ssDNA derived from extracted total RNA; the amplicons were the appropriate size when assessed via gel in our lab while the sequencing facility found primer contamination during their Bioanalyzer trace/QC steps. They ran a cleanup on the library pool upon my request. The remaining 44 samples (amplicon templates were regular DNA) on the run turned out to be high quality throughout the 300 cycles for both F and R. This leads me to believe that the failure point was somewhere on my end of things. 

Unless @benjjneb has anything to add in, I think it's ok to close this (non-dada) issue. I'll reply to this thread if any in-silico solution is found. 

Thank you once again for your help. 

JP

**Comment by benjjneb** [comment]
I agree with @joey711 that this looks like something went wrong, so not much to add.

In the future though, more permissive quality filtering parameters (eg. `maxEE=6`) are a valid approach with dada2. Because the quality scores are being modeled, dada2 is not sensitive to the filtering params the way OTU methods are. It will cost you in computation time, but that is not usually a blocking problem.

**Comment by joey711** [comment]
@benjjneb just out of curiosity, have you looked into any possible positional effect on the model? Ignoring for a moment how much more complicated this would be to address in practice, how confident are you that the increased error-rate at the end of sequences is just that, simply an increase in rate and not a change in the underlying error-process? I think I remember that you had done so, at least in regions, but it seems apropos to this discussion, too, since it motivates why one might not want to try to use data that exhibits the artifacts in this post.

**Comment by JacobRPrice** [comment]
@joey711 your comment was one of the things that came up in some of our lab group's discussions about the dada2 methodology. 

The 6 MiSeq runs I have access to (a limited sample size, I know) all indicate that quality monotonically decreases as the run progresses, and that there's generally a break point where the slope severely changes. I assumed that the error profile would be different for the first, say 220 cycles, versus the last 30 cycles on a 250 bp run. I'm not sure if this is the most ideal or correct way to carry it out, but I've been trimming my reads to minimize the "mixing" of error profiles while still having a minimum of 20 bp overlap.

**Comment by benjjneb** [comment]
The positional independence assumption (after accounting for quality score) has fit quite well when I've tested it -- but I've tested it in reasonably well-behaved sequencing runs (i.e. a fairly smooth decline in the quality profile).

There's reason to think it will still work pretty well even in runs like this that have a break point from good to bad quality, in part because the estimation of the high and low quality parts will be largely independent (error rates over the quality scores are estimated by a loess fit, which is only locally linked).

However, that is supposition at this point, and I think @JacobRPrice approach of trimming the sequence after the crash, provided there is enough overlap, is a prudent one.

**Comment by joey711** [comment]
@JacobRPrice , to emphasize @benjjneb point, make sure to be careful not to conflate a change in Err-v-Q model (dependence of actual error rate on observed quality value) with a change in the average positional quality at a position. The fact that average quality declines at later positions in the run is not evidence that Err-v-Q model has changed at all, only that lower average Q values are being reported by the machine. 

The thing to look for would be shifts in the Err-v-Q-model when fit exclusively for different positions or regions of reads from the same run. The stellar DADA2 performance on validation data is already de facto evidence that the positional-independence model is approximately true, in the sense that it is more than good enough to describe good data. Our motivation for questioning it in this case is the hypothesis that a large "cliff" or other non-smooth change in average Q-by-position is a symptom of an underlying change in the sequencing process of less-than-good-data, which *might* also coincide with a shift in the Err-v-Q-model because of said change. If true, it would motivate additional heuristics for selecting trimming and filtering parameters for datasets with this feature. Again, this hypothesis doesn't really implicate "smooth/nice" looking data, only suggests that looking more deeply at how well the positional-independence assumption holds in these different special-case scenarios might be interesting, or worth considering in parameter choices.

**Comment by janetw** [comment]
@JacobRPrice Sorry to jump in on this thread but as I was searching for help on merging poor quality read data, I came upon this posting. I am working on a sequencing data set from a MiSeq 2x300 bp run that has very similar read quality profiles. There is a drop-off right around 100. My first thought was that it was some sort of a sequencing issue since this was the first time I had seen this. But I talked with our core and the quality over the whole run does not look bad so now I am thinking it is likely something on our end. Were you ever able to figure out what was the cause or reason for this? Thanks for any insight. 
- Janet

**Comment by JacobRPrice** [comment]
@janetw I re-tested the quality of the rRNA (and cDNA product). The rRNA wasn't the cleanest looking on nanodrop and the cDNA was worse. With the number of people who use the EMP primers I can't say that they or their protocols are off; I believe something was done incorrectly or a step failed somewhere along the rRNA --> DNase --> cDNA synthesis pathway and the PCR step failed or at least had a high degree of error. This is supported by the other part of the run (identical samples prepped from rDNA) had very high quality throughout. 

It's a little aggravating, but I've abandoned trying to use both forward and reverse. I just used the forward reads, set EE to 6, and fiddled with the trimming so I was able to keep a large portion of the reads. As Ben stated above, the cost of having a high EE (and high error) is computation time. 

I hope this helps!
